src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCheck.java

```diff
@@ -70,6 +84,17 @@
  *   &lt;property name=&quot;allowLineBreaks&quot; value=&quot;true&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * Map .Entry e = null; // violation, whitespace before '.' is not allowed
+ * Map.Entry f = null; // OK
+ * int foo
+ * ; // OK, linebreak is allowed
+ * Customer c = new Customer(); // Customer is a user-defined class
+ * c.setFirstName("John")
+ *  .setLastName("Doe")
+ *  ; // OK, linebreak is allowed
```

Comment:
this is ok because of token target is only DOT. fix the same for `foo`. in this example.

Reply:
Actually, this line of code serve the purpose of showing the difference between default setup and setup with the property of allowLineBreaks to true. Should I think of some different example for demonstrating the difference between the two configurations.

Reply:
```
c.setFirstName("John")
 .setLastName("Doe");
```
It is OK with default setup too, the only 

> @Abhishek-kumar09 ,
> 
> in your example:
> 
> ```
> 	c.setFirstName("John") // OK
> 	 .setLastName("Doe")  // violation
> 	 .setPhone("+1 1234567890")
> 	 ; // violation, linebreak is not allowed by default
> ```
> 
> I do not see violation over `.setLastName` from CLI, please recheck and update description examples

Sorry, it remained there. I will fix it.

Reply:
This line of code serves the purpose of differentiating between the two configurations (deafult and allowLineBreaks), Should I think of some different example to show the difference between two configurations.

Reply:
to make it for user to show change in behavior we need to do following
1) show default config with code 
2) show config with allowLineBreaks enabled only, show how behavior is changing
3) show config with modified `tokens` to show that violations could be avoided from previous example by defining target of validation.

you can get inspiration on custom targets at https://github.com/checkstyle/checkstyle/blob/aa5a9fd19a6c2e05dbfa756e259f7f59b7d676ce/src/main/resources/google_checks.xml#L252
and at https://github.com/checkstyle/checkstyle/blob/646d90e0237d92cb6ab84dc7083a1d2d636eceb2/config/checkstyle_checks.xml#L709

Reply:
the reason of default is ok with chained methods because it does not validate them, DOT is not default token - https://checkstyle.org/config_whitespace.html#NoWhitespaceBefore .

Lets make example 2 with all possible tokens defined, to show show difference in behavior for each token with and wihtout allowLineBreaks. So user will get clear picture of how it works.

Reply:
Is it OK to have three configuration like:
NoWhitespaceBefore - default
NoWhitespaceBefore - with only DOT token
NoWhitespaceBefore - with DOT and allowedLineBreaks

Reply:
Sorry for being late there, I had some personal issues, I have pushed the code with three configuration
```
1. with default configuration
2. with allowLineBreak on default tokens
3. with allowLineBreak on only DOT token
```
thankyou @romani